### ISSUE TITLE
"sudo apt update" causes leaks.

### ISSUE THREAD

**Issue by aascedu** [42-tronc/minishell#201]
```minishell> sudo apt update                                                                              
==21343== 
==21343== Warning: Can't execute setuid/setgid/setcap executable: /usr/bin/sudo
==21343== Possible workaround: remove --trace-children=yes, if in effect
==21343== 
==21343== 
==21343== FILE DESCRIPTORS: 3 open at exit.
==21343== Open file descriptor 2: /dev/pts/0
==21343==    <inherited from parent>
==21343== 
==21343== Open file descriptor 1: /dev/pts/0
==21343==    <inherited from parent>
==21343== 
==21343== Open file descriptor 0: /dev/pts/0
==21343==    <inherited from parent>
==21343== 
==21343== 
==21343== HEAP SUMMARY:
==21343==     in use at exit: 196,142 bytes in 211 blocks
==21343==   total heap usage: 801 allocs, 590 frees, 232,697 bytes allocated
==21343== 
==21343== 11 bytes in 2 blocks are still reachable in loss record 4 of 50
==21343==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==21343==    by 0x409116: ft_strdup (in /home/ar7/Documents/minishell/minishell)
==21343==    by 0x401B69: get_cmd_args (execve.c:43)
==21343==    by 0x401EAF: execve_cmd (execve.c:138)
==21343==    by 0x404ABF: create_subshell (utils_subshell.c:78)
==21343==    by 0x40542B: check_command (minishell.c:57)
==21343==    by 0x405105: exec_dispatch (minishell.c:103)
==21343==    by 0x40558A: main (minishell.c:137)
==21343== 
==21343== 14 bytes in 1 blocks are still reachable in loss record 6 of 50
==21343==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==21343==    by 0x409116: ft_strdup (in /home/ar7/Documents/minishell/minishell)
==21343==    by 0x401AD7: get_cmd_args (execve.c:36)
==21343==    by 0x401EAF: execve_cmd (execve.c:138)
==21343==    by 0x404ABF: create_subshell (utils_subshell.c:78)
==21343==    by 0x40542B: check_command (minishell.c:57)
==21343==    by 0x405105: exec_dispatch (minishell.c:103)
==21343==    by 0x40558A: main (minishell.c:137)
==21343== 
==21343== 32 bytes in 1 blocks are still reachable in loss record 11 of 50
==21343==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==21343==    by 0x4091FD: ft_calloc (in /home/ar7/Documents/minishell/minishell)
==21343==    by 0x401AA6: get_cmd_args (execve.c:32)
==21343==    by 0x401EAF: execve_cmd (execve.c:138)
==21343==    by 0x404ABF: create_subshell (utils_subshell.c:78)
==21343==    by 0x40542B: check_command (minishell.c:57)
==21343==    by 0x405105: exec_dispatch (minishell.c:103)
==21343==    by 0x40558A: main (minishell.c:137)
==21343== 
==21343== LEAK SUMMARY:
==21343==    definitely lost: 0 bytes in 0 blocks
==21343==    indirectly lost: 0 bytes in 0 blocks
==21343==      possibly lost: 0 bytes in 0 blocks
==21343==    still reachable: 57 bytes in 4 blocks
==21343==         suppressed: 196,085 bytes in 207 blocks
==21343== 
==21343== For lists of detected and suppressed errors, rerun with: -s
==21343== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

**Comment by aascedu** [comment]
with valgrind option --trace-children=yes

**Comment by aascedu** [comment]
any command launched with sudo before will make it leaks :

**Comment by aascedu** [comment]
```
minishell> sudo ls
==57992== 
==57992== Warning: Can't execute setuid/setgid/setcap executable: /usr/bin/sudo
==57992== Possible workaround: remove --trace-children=yes, if in effect
==57992== 
==57992== 
==57992== FILE DESCRIPTORS: 3 open at exit.
==57992== Open file descriptor 2: /dev/pts/0
==57992==    <inherited from parent>
==57992== 
==57992== Open file descriptor 1: /dev/pts/0
==57992==    <inherited from parent>
==57992== 
==57992== Open file descriptor 0: /dev/pts/0
==57992==    <inherited from parent>
==57992== 
==57992== 
==57992== HEAP SUMMARY:
==57992==     in use at exit: 196,126 bytes in 210 blocks
==57992==   total heap usage: 758 allocs, 548 frees, 231,798 bytes allocated
==57992== 
==57992== 3 bytes in 1 blocks are still reachable in loss record 1 of 50
==57992==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==57992==    by 0x409116: ft_strdup (in /home/ar7/Documents/minishell/minishell)
==57992==    by 0x401B69: get_cmd_args (execve.c:43)
==57992==    by 0x401EAF: execve_cmd (execve.c:138)
==57992==    by 0x404ABF: create_subshell (utils_subshell.c:78)
==57992==    by 0x40542B: check_command (minishell.c:57)
==57992==    by 0x405105: exec_dispatch (minishell.c:103)
==57992==    by 0x40558A: main (minishell.c:137)
==57992== 
==57992== 14 bytes in 1 blocks are still reachable in loss record 6 of 50
==57992==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==57992==    by 0x409116: ft_strdup (in /home/ar7/Documents/minishell/minishell)
==57992==    by 0x401AD7: get_cmd_args (execve.c:36)
==57992==    by 0x401EAF: execve_cmd (execve.c:138)
==57992==    by 0x404ABF: create_subshell (utils_subshell.c:78)
==57992==    by 0x40542B: check_command (minishell.c:57)
==57992==    by 0x405105: exec_dispatch (minishell.c:103)
==57992==    by 0x40558A: main (minishell.c:137)
==57992== 
==57992== 24 bytes in 1 blocks are still reachable in loss record 9 of 50
==57992==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==57992==    by 0x4091FD: ft_calloc (in /home/ar7/Documents/minishell/minishell)
==57992==    by 0x401AA6: get_cmd_args (execve.c:32)
==57992==    by 0x401EAF: execve_cmd (execve.c:138)
==57992==    by 0x404ABF: create_subshell (utils_subshell.c:78)
==57992==    by 0x40542B: check_command (minishell.c:57)
==57992==    by 0x405105: exec_dispatch (minishell.c:103)
==57992==    by 0x40558A: main (minishell.c:137)
==57992== 
==57992== LEAK SUMMARY:
==57992==    definitely lost: 0 bytes in 0 blocks
==57992==    indirectly lost: 0 bytes in 0 blocks
==57992==      possibly lost: 0 bytes in 0 blocks
==57992==    still reachable: 41 bytes in 3 blocks
==57992==         suppressed: 196,085 bytes in 207 blocks
==57992== 
==57992== For lists of detected and suppressed errors, rerun with: -s
==57992== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

**Comment by maplepy** [comment]
fix in 071f858996e1e9a5e5d34d7f561f3c26a332fb63
